### PR TITLE
Add new `RSpec/Capybara/PredicateMatcher` cop

### PR DESCRIPTION
**Pull request by ydah**
Follow up: https://github.com/rubocop/rubocop-rspec/pull/1456/files#r1026096951

______________________________________________________________________

Before submitting the PR make sure the following are checked:

- [x] Feature branch is up-to-date with `master` (if not - rebase it).
- [x] Squashed related commits together.
- [x] Added tests.
- [-] Updated documentation.
- [x] Added an entry to the `CHANGELOG.md` if the new code introduces user-observable changes.
- [x] The build (`bundle exec rake`) passes (be sure to run this locally, since it may produce updated documentation that you will need to commit).

### CODE REVIEW

File: spec/rubocop/cop/rspec/predicate_matcher_spec.rb [hunk 1]
```diff
@@ -64,6 +66,7 @@
           expect(foo).to include(something)
           expect(foo).to respond_to(:bar)
           expect(foo).to be_something()
+          expect(foo).to match_style(bar: 'baz')
```
Comment (pirj) [inline spec/rubocop/cop/rspec/predicate_matcher_spec.rb:69]: Would this work with any method?
`rspec-expectations` magically provides `have_` and `be_` dynamic predicate matchers, but I suspect not `match_` ones.
In this case, it's an existing matcher defined by capybara. What if such a matcher doesn't exist, e.g.:

```ruby
expect(inventory.matches_or_a_lighter?).to be_truthy
```

Comment (ydah) [reply spec/rubocop/cop/rspec/predicate_matcher_spec.rb]: I remembered the plan that from now on Capybara related cops will be separated from RuboCop RSpec, so I thought again in the direction of creating a new cop in `RSpec/Capybara/*`. What do you think?

Comment (pirj) [reply spec/rubocop/cop/rspec/predicate_matcher_spec.rb]: Makes sense :+1:

**Review by pirj — APPROVED** [review]
Nice refactoring and a cool cop, thank you!

**Comment by ydah** [comment]
It was conflicted, so I rebased it to the latest master branch.

**Review by bquorning — COMMENTED** [review]
Won’t this PR make it harder to split e.g. the Capybara cops into a separate gem? See https://github.com/rubocop/rubocop-rspec/discussions/1440 for context.

**Comment by ydah** [comment]
I updated this PR for easy division with reference to https://github.com/rubocop/rubocop-rspec/pull/1511 .

File: lib/rubocop/cop/rspec/capybara/predicate_matcher.rb [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        #   # good
+        #   expect(foo).to match_css(bar: 'baz')
+        #   expect(foo).to match_selector(bar: 'baz')
+        #   expect(foo).to match_style(bar: 'baz')
+        #   expect(foo).to match_xpath(bar: 'baz')
+        #
+        #   # also good - It checks "true" strictly.
+        #   expect(foo.matches_style?(bar: 'baz')).to be(true)
+        #
+        # @example Strict: false, EnforcedStyle: inflected
```
[the remark this answers is not in the document]
Comment (ydah) [reply lib/rubocop/cop/rspec/capybara/predicate_matcher.rb:241]: The default is to have the same behavior as `RSpec/PredicateMatcher`.
https://github.com/rubocop/rubocop-rspec/blob/e0ff0a0f45c3d53d430378d2336cde2a4260bedf/config/default.yml#L683-L687

However, I personally do not have strong feelings that things should remain as they are.

**Comment by ydah** [comment]
It was conflicted, so I rebased it to the latest master branch.

**Comment by pirj** [comment]
Please don't merge this just yet, I'm extracting `rspec-capybara`.

**Comment by pirj** [comment]
Wondering if `rubocop-rspec` or `rubocop-capybara` is the best fit for this cop 🤔

**Comment by ydah** [comment]
It is indeed difficult to find this cop 🤔 .
RSpec, Minitest and more framework when managed in rubocop-capybara and each testing framework must be able to handle it.
But, I think it is worth to consider a good way to support each testing framework. WDYT?

**Comment by pirj** [comment]
I suggest re-targeting this PR to `rspec-capybara`. We planned to have support for Minitest there, too, but we have some RSpec-only cops now there too.

We only kept `FeatureMethods`, which I would actually move to the root `RSpec` department renaming it to something like `UnnecessaryExampleGroupAlias`.

**Comment by ydah** [comment]
I have opened the following PR and will close this PR
- https://github.com/rubocop/rubocop-capybara/pull/55

**Pull request closed without merging**